app/controllers/users_controller.rb

```diff
@@ -11,10 +13,20 @@ def show
   def new
   end
 
+  def activate
+    if @user = User.load_from_activation_token(params[:id])
+      @user.activate!
+      auto_login(@user)
+      redirect_to(user_groups_path, notice: "Welcome #{@user.name}! Start by creating your first group of people you're shopping for.")
+    else
+      flash[:notice] = 'Invalid confirmation token.'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.